flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesLabel.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.utils;
+
+import org.apache.flink.kubernetes.configuration.KubernetesConfigOptions;
+
+import java.util.Arrays;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.StringUtils.isNullOrWhitespaceOnly;
+
+/** Collection of kubernetes labels and helper methods. */
+public enum KubernetesLabel {
+    CONFIG_MAP_PREFIX("flink-config-", ""),
+    HADOOP_CONF_CONFIG_MAP_PREFIX("hadoop-config-", ""),
+    KERBEROS_KEYTAB_SECRET_PREFIX("kerberos-keytab-", ""),
+    KERBEROS_KRB5CONF_CONFIG_MAP_PREFIX("kerberos-krb5conf-", ""),
+    POD_TEMPLATE_CONFIG_MAP_PREFIX("pod-template-", ""),
+    FLINK_REST_SERVICE_SUFFIX("", "-rest");
+
+    private final String prefix;
+    private final String suffix;
+
+    KubernetesLabel(String prefix, String suffix) {
+        this.prefix = prefix;
+        this.suffix = suffix;
+    }
+
+    private int getTotalLength() {
+        return prefix.length() + suffix.length();
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
+    public static int getClusterIdMaxLength() {
+        int longestAffix =
+                Arrays.stream(KubernetesLabel.values())
+                        .map(KubernetesLabel::getTotalLength)
+                        .max(Integer::compareTo)
+                        .get();
```

Comment:
```suggestion
                        .orElseThrow(() -> new IllegalStateException("No enum value is present."));
```
nit: you could remove this method's `@SuppressWarnings` annotation by stating the actual issue in the code. That makes the issue more explicit.